src/render.tsx

```diff
@@ -25,17 +28,65 @@ export type RenderResult = ReturnType<typeof render>;
  */
 export default function render<T>(
   component: React.ReactElement<T>,
-  { wrapper: Wrapper, createNodeMock }: RenderOptions = {}
+  {
+    wrapper: Wrapper,
+    createNodeMock,
+    unstable_validateStringsRenderedInText,
+  }: RenderOptions = {}
 ) {
-  const wrap = (innerElement: React.ReactElement) =>
-    Wrapper ? <Wrapper>{innerElement}</Wrapper> : innerElement;
+  if (unstable_validateStringsRenderedInText) {
+    return renderWithStringValidation(component, {
+      wrapper: Wrapper,
+      createNodeMock,
+    });
+  }
+
+  const wrap = (element: React.ReactElement) =>
+    Wrapper ? <Wrapper>{element}</Wrapper> : element;
 
   const renderer = renderWithAct(
     wrap(component),
     createNodeMock ? { createNodeMock } : undefined
   );
+
+  return enrichRenderResult(renderer, wrap);
```

Comment:
`enrich` naming here is too generic, maybe something along the line of:
```suggestion
  return buildRenderResult(renderer, wrap);
```